### ISSUE TITLE
Grove Sprite Server Crash

### ISSUE THREAD

**Issue by lisosaurus** [Daveyx0/MultiMob#3]
Might be related to #1. Every once in a while, my server will crash with the following error:

```
java.lang.NoClassDefFoundError: net/daveyx0/multimob/util/ColorUtil
	at net.daveyx0.primitivemobs.entity.passive.EntityGroveSprite.getColor(EntityGroveSprite.java:286)
	at net.daveyx0.primitivemobs.item.ItemGroveSpriteSap.onGroveSpriteDrop(ItemGroveSpriteSap.java:55)
	at net.daveyx0.primitivemobs.entity.passive.EntityGroveSprite.func_70628_a(EntityGroveSprite.java:387)
	at net.minecraft.entity.EntityLivingBase.func_184610_a(EntityLivingBase.java:1217)
	at net.minecraft.entity.EntityLiving.func_184610_a(EntityLiving.java:550)
	at net.minecraft.entity.EntityLivingBase.func_70645_a(EntityLivingBase.java:1197)
	at net.minecraft.entity.EntityLivingBase.func_70097_a(EntityLivingBase.java:1023)
	at net.daveyx0.primitivemobs.entity.passive.EntityGroveSprite.func_70097_a(EntityGroveSprite.java:220)
	at net.minecraft.entity.EntityLivingBase.func_70030_z(EntityLivingBase.java:234)
	at net.minecraft.entity.EntityLiving.func_70030_z(EntityLiving.java:208)
	at net.minecraft.entity.Entity.func_70071_h_(Entity.java:389)
	at net.minecraft.entity.EntityLivingBase.func_70071_h_(EntityLivingBase.java:2096)
	at net.minecraft.entity.EntityLiving.func_70071_h_(EntityLiving.java:295)
	at net.daveyx0.primitivemobs.entity.passive.EntityGroveSprite.func_70071_h_(EntityGroveSprite.java:208)
	at net.minecraft.world.World.func_72866_a(World.java:1993)
	at net.minecraft.world.WorldServer.func_72866_a(WorldServer.java:832)
	at net.minecraft.world.World.func_72870_g(World.java:1955)
	at net.minecraft.world.World.func_72939_s(World.java:1759)
	at net.minecraft.world.WorldServer.func_72939_s(WorldServer.java:613)
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:767)
	at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:397)
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:668)
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:526)
	at java.lang.Thread.run(Unknown Source)
Caused by: java.lang.ClassNotFoundException: net.daveyx0.multimob.util.ColorUtil
	at net.minecraft.launchwrapper.LaunchClassLoader.findClass(LaunchClassLoader.java:191)
	at java.lang.ClassLoader.loadClass(Unknown Source)
	at java.lang.ClassLoader.loadClass(Unknown Source)
	... 24 more
Caused by: net.minecraftforge.fml.common.asm.ASMTransformerWrapper$TransformerException: Exception in class transformer net.minecraftforge.fml.common.asm.transformers.SideTransformer@5b12012e from coremod FMLCorePlugin
	at net.minecraftforge.fml.common.asm.ASMTransformerWrapper$TransformerWrapper.transform(ASMTransformerWrapper.java:260)
	at net.minecraft.launchwrapper.LaunchClassLoader.runTransformers(LaunchClassLoader.java:279)
	at net.minecraft.launchwrapper.LaunchClassLoader.findClass(LaunchClassLoader.java:176)
	... 26 more
Caused by: java.lang.RuntimeException: Attempted to load class net/daveyx0/multimob/util/ColorUtil for invalid side SERVER
	at net.minecraftforge.fml.common.asm.transformers.SideTransformer.transform(SideTransformer.java:62)
	at net.minecraftforge.fml.common.asm.ASMTransformerWrapper$TransformerWrapper.transform(ASMTransformerWrapper.java:256)
	... 28 more
```

This seems to happen at random, as nobody is doing anything with the grove sprites when it happens. It only occurs when players are online, however.

**Comment by Daveyx0** [comment]
Somehow the Grove Sprite is trying to use GetColor on the Server side; but that makes no sense.... it specifically checks if it's client.... will check but Im confused

**Comment by Daveyx0** [comment]
Im having trouble reproducing this. Just ran a server and spawned like 100 of em. These kinda bugs are the tricky ones....

**Comment by Daveyx0** [comment]
I'm also using that one. Are all your players up to date? In the next update I made a small change but that should not affect this much..... hmmm

**Comment by Daveyx0** [comment]
Let me know if it still happens with the most recent version :)

**Comment by lisosaurus** [comment]
After the latest update, it's now crashing with a slightly different error:
`java.lang.NoSuchMethodError: net.daveyx0.primitivemobs.entity.passive.EntityGroveSprite.getColor(Lnet/minecraft/world/World;Lnet/minecraft/block/state/IBlockState;Lnet/minecraft/util/math/BlockPos;Lnet/minecraft/util/EnumFacing;)[I
	at net.daveyx0.primitivemobs.item.ItemGroveSpriteSap.onGroveSpriteDrop(ItemGroveSpriteSap.java:55)
	at net.daveyx0.primitivemobs.entity.passive.EntityGroveSprite.func_70628_a(EntityGroveSprite.java:394)
	at net.minecraft.entity.EntityLivingBase.func_184610_a(EntityLivingBase.java:1217)
	at net.minecraft.entity.EntityLiving.func_184610_a(EntityLiving.java:550)
	at net.minecraft.entity.EntityLivingBase.func_70645_a(EntityLivingBase.java:1197)
	at net.minecraft.entity.EntityLivingBase.func_70097_a(EntityLivingBase.java:1023)
	at net.daveyx0.primitivemobs.entity.passive.EntityGroveSprite.func_70097_a(EntityGroveSprite.java:226)
	at net.minecraft.entity.EntityLivingBase.func_70030_z(EntityLivingBase.java:234)
	at net.minecraft.entity.EntityLiving.func_70030_z(EntityLiving.java:208)
	at net.minecraft.entity.Entity.func_70071_h_(Entity.java:389)
	at net.minecraft.entity.EntityLivingBase.func_70071_h_(EntityLivingBase.java:2096)
	at net.minecraft.entity.EntityLiving.func_70071_h_(EntityLiving.java:295)
	at net.daveyx0.primitivemobs.entity.passive.EntityGroveSprite.func_70071_h_(EntityGroveSprite.java:214)
	at net.minecraft.world.World.func_72866_a(World.java:1993)
	at net.minecraft.world.WorldServer.func_72866_a(WorldServer.java:832)
	at net.minecraft.world.World.func_72870_g(World.java:1955)
	at net.minecraft.world.World.func_72939_s(World.java:1759)
	at net.minecraft.world.WorldServer.func_72939_s(WorldServer.java:613)
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:767)
	at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:397)
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:668)
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:526)
	at java.lang.Thread.run(Unknown Source)`

This is running both the latest Primitive Mobs (1.2 Beta) and MultiMob (1.0.2).

**Comment by Daveyx0** [comment]
I found the issue and resolved it in 1.2.2 ;)
